ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -65,104 +71,158 @@ def init_regex_and_parser(self, data_directory_path):
             number_list (list): list containing updated detected numeric text
             original_list (list): list containing updated original numeral text
+
+        Examples:
+            [In]  >>  self.processed_text = "One hundred two"
+            [In]  >>  _detect_number_from_numerals()
+            [Out] >> ([{'value': '102', 'unit': None}], ['one hundred two two'])
+
+            [In]  >>  self.processed_text = "two hundred - three hundred"
+            [In]  >>  _detect_number_from_numerals()
+            [Out] >> ([{'value': '200', 'unit': None}, {'value': '300', 'unit': None}],
+                      ['two hundred', 'three hundred'])
+
+            [In]  >>  self.processed_text = "one two three"
+            [In]  >>  _detect_number_from_numerals()
+            [Out] >> ([{'value': '2', 'unit': None}, {'value': '2', 'unit': None}, {'value': '3', 'unit': None}],
+                      ['one', 'two', 'three'])
         """
         number_list = number_list or []
         original_list = original_list or []
 
-        current = result = 0
-        current_text, result_text = '', ''
-        on_number = False
-        for word in self.processed_text.split():
-            if word not in self.numbers_word:
-                if on_number:
-                    original = (result_text.strip() + " " + current_text.strip()).strip()
-                    number_detected = int(result + current) if float(result + current).is_integer() \
-                        else result + current
-                    if self.max_digit >= len(str(number_detected)) >= self.min_digit:
-                        number_list.append(str(number_detected))
-                        original_list.append(original)
-
-                result = current = 0
-                result_text, current_text = '', ''
-                on_number = False
-            else:
-                scale, increment = self.numbers_word[word]
-                # handle where only scale is mentioned without unit, for ex - thousand(for 1000), hundred(for 100)
-                current = 1 if(scale > 0 and current == 0 and increment == 0) else current
-                current = current * scale + increment
-                current_text = (current_text + " " + word).strip()
-                if scale > 100:
-                    result += current
-                    result_text = (result_text + " " + current_text).strip()
-                    current = 0
-                    current_text = ''
-                on_number = True
-
-        if on_number:
-            original = (result_text.strip() + " " + current_text.strip()).strip()
-            number_detected = int(result + current) if float(result + current).is_integer() else result + current
-            if self.max_digit >= len(str(number_detected)) >= self.min_digit:
-                number_list.append(str(number_detected))
-                original_list.append(original)
-
+        numeral_text_list = re.split(r'[\-\:]', self.processed_text)
```

Comment:
Can we add a comment to explain why we are tokenizing this way? I am unable to understand why this is being done but I am sure there is a good reason

Reply:
Splitting text based on "-" and ":",  as in case of text "two thousand-three thousand", simple splitting will give list as [two, thousand-three, thousand], result in number word detector giving wrong result, 
hence we need to separate them into [two thousand, three thousand] using '-' or ':' as split char

>> Added same in comment above this line